tests/testthat/test-html_document.R

```diff
@@ -0,0 +1,22 @@
+test_that("anchor_sections adds correct component", {
+  skip_if_not(pandoc2.0())
+  expect_equivalent(add_anchor_sections(FALSE), list(NULL, NULL, NULL))
```

Comment:
This would be a good place for snapshot tests, if you can use them.

Reply:
I am just looking into testthat 3 for rmarkdown. I think it is important so yes I hope I can use them. 

I understand `expect_equivalent` is deprecated, so I should change it.

Reply:
I'll introduce testthat 3 later and not introduce it in this PR. I change for `expect_equal` here.